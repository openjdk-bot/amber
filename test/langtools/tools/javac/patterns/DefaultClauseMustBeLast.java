/*
 * @test /nodynamiccopyright/
 * @summary Ensure that in a matching switch statement, the default clause must be the last.
 * @compile/fail/ref=DefaultClauseMustBeLast.out -XDrawDiagnostics DefaultClauseMustBeLast.java
 */
public class DefaultClauseMustBeLast {
    public static void main(String[] args) {
        Object o = (Long)42L;
        switch (o) {
            default: System.out.println("default"); break;
            case Long l: System.out.println("It's a long"); break;
        }
        switch (o) {
            case Long l: System.out.println("It's a long"); break;
            default : System.out.println("default"); break;
            case Double d: System.out.println("It's a double"); break;
        }
    }
}
