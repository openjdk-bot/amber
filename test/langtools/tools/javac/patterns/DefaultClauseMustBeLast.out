DefaultClauseMustBeLast.java:11:13: compiler.err.pattern.dominated
DefaultClauseMustBeLast.java:16:13: compiler.err.pattern.dominated
2 errors
